storage/replica.go

```diff
@@ -1183,21 +1184,25 @@ func (r *Replica) handleRaftReady() error {
 
 func (r *Replica) sendRaftMessage(msg raftpb.Message) {
 	groupID := r.RangeID
-	r.store.mu.Lock()
-	toReplica, err := r.store.replicaDescriptorLocked(groupID, roachpb.ReplicaID(msg.To))
-	if err != nil {
-		log.Warningf("failed to lookup recipient replica %d in group %s: %s", msg.To, groupID, err)
-		r.store.mu.Unlock()
-		return
-	}
-	fromReplica, err := r.store.replicaDescriptorLocked(groupID, roachpb.ReplicaID(msg.From))
-	if err != nil {
-		log.Warningf("failed to lookup sender replica %d in group %s: %s", msg.From, groupID, err)
-		r.store.mu.Unlock()
+	var toReplica, fromReplica roachpb.ReplicaDescriptor
+	if err := func() error {
```

Comment:
I don't like functions defined inline in an if statement, and in general this case seems to be adding more complexity than it's worth. I don't think we need to ensure that all unlocks happen via defer. But if you want to get rid of the early returns from the critical section here, I would suggest

```
r.store.mu.Lock()
toReplica, toErr := r.store.replicaDescriptorLocked(...)
fromReplica, fromErr := r.store.replicaDescriptorLocked(...)
r.store.mu.Unlock()
if toErr != nil {
  log.Warningf(...)
  return
}
if fromErr != nil {
  log.Warningf(...)
  return
}
```